File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.js


function solution(nums) {
    n = nums.length / 2;
    
    keys = [];
    
    for(const num of nums) {
        if (!keys.find(key => key == num)) {
            keys.push(num);
            if (keys.length >= n) {
                return n;
            }
        };
    }
    return keys.length;
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int n = nums.size() / 2;
    
    vector<int> v;
    
    for (const int num: nums) {
        if (find(v.begin(), v.end(), num) == v.end()) {
            v.push_back(num);
        }
        
        if (v.size() >= n) {
            return n;
        }
    }
    
    return v.size();
}